src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,243 @@
+        } catch (NullPointerException e) {
+            System.out.printf("Error: Task %d does not exist", index);
+        } catch (IndexOutOfBoundsException e) {
+            System.out.printf("Error: Task %d does not exist", index);
+        }
+    }
+
+    private static void printDoneCmdHelp() {
+        printResponse("Usage: done [task index]");
+        printResponse("Example: done 1");
+    }
+
+    private static void handleAddTodo(String cmd) {
+        int descriptionIndex = TODO_COMMAND.length();
+        String description = cmd.substring(descriptionIndex);
+        try {
+            Todo todo = new Todo(description);
+            addTask(todo);
+        } catch (EmptyDescriptionException e) {
+            printTodoCmdHelp();
+        }
+    }
+
+    private static void printTodoCmdHelp() {
+        printResponse("Usage: todo [description]");
+        printResponse("Example: todo borrow book");
+    }
+
+    private static void handleAddDeadline(String cmd) {
+        int descriptionIndex = DEADLINE_COMMAND.length();
+        int dateIndex = cmd.indexOf(BY_ARGUMENT);
+        String description, date;
+
+        try {
+            description = cmd.substring(descriptionIndex, dateIndex - 1);
+            date = cmd.substring(dateIndex + BY_ARGUMENT.length());
+        } catch (StringIndexOutOfBoundsException e) {
+            printDeadlineCmdHelp();
+            return;
+        }
+
+        try {
+            Deadline deadline = new Deadline(description, date);
+            addTask(deadline);
+        } catch (EmptyDescriptionException e) {
+            printTodoCmdHelp();
+        }
+    }
+
+    private static void printDeadlineCmdHelp() {
+        printResponse("Usage: deadline [description] /by [when]");
+        printResponse("Example: deadline review pr /by tomorrow");
+    }
+
+    private static void handleAddEvent(String cmd) {
+        int descriptionIndex = EVENT_COMMAND.length();
+        int dateIndex = cmd.indexOf(AT_ARGUMENT);
+        String description, date;
+
+        try {
+            description = cmd.substring(descriptionIndex, dateIndex - 1);
+            date = cmd.substring(dateIndex + AT_ARGUMENT.length());
+        } catch (StringIndexOutOfBoundsException e) {
+            printEventCmdHelp();
+            return;
+        }
+
+        try {
+            Event event = new Event(description, date);
+            addTask(event);
+        } catch (EmptyDescriptionException e) {
+            printTodoCmdHelp();
+        }
+    }
+
+    private static void printEventCmdHelp() {
+        printResponse("Usage: event [description] /at [when]");
+        printResponse("Example: event presentation /at tomorrow");
+    }
+
+    private static void addTask(Task task) {
+        tasks[taskCount++] = task;
+        printResponse(String.format("Added: %s", task));
+        printResponse(String.format("You now have %d task%s in your list.", taskCount, (taskCount == 1 ? "" : "s")));
+    }
```

Comment:
Did you check if your taskCount exceeds your task capacity? And the use ternary operator makes the logic of the code hard to read.

Reply:
Fixed, thanks.